test/end-to-end/transactionType/transactionType.spec.js

```diff
@@ -0,0 +1,119 @@
+    text: 'E2E Other Transaction Type',
+    description: 'Test Other transaction type',
+    type: 'other',
+    prefix: 'E2EOTT',
+    other: 'OTHER_TYPE'
+  };
+
+  it('Successfully creates a transaction type', () => {
+    FU.buttons.create();
+    FU.input('$ctrl.transactionType.text', newType.text);
+    FU.input('$ctrl.transactionType.description', newType.description);
+    FU.input('$ctrl.transactionType.prefix', newType.prefix);
+    element(by.model('$ctrl.transactionType.type')).click();
+    element(by.css(`[value=${newType.type}]`)).click();
+    FU.buttons.submit();
+    components.notification.hasSuccess();
+  });
+
+  it('Successfully updates an existing transaction type', () => {
+    element(by.css('[data-edit-type="' + identifier + '"]')).click();
+    FU.input('$ctrl.transactionType.text', updateType.text);
+    FU.input('$ctrl.transactionType.description', updateType.description);
+    FU.input('$ctrl.transactionType.prefix', updateType.prefix);
+    element(by.model('$ctrl.transactionType.type')).click();
+    element(by.css(`[value=${updateType.type}]`)).click();
+    FU.buttons.submit();
+    components.notification.hasSuccess();
+  });
+
+  it('Successfully creates a transaction type with a specific type', () => {
+    FU.buttons.create();
+    FU.input('$ctrl.transactionType.text', otherType.text);
+    FU.input('$ctrl.transactionType.description', otherType.description);
+    FU.input('$ctrl.transactionType.prefix', otherType.prefix);
+    element(by.model('$ctrl.transactionType.type')).click();
+    element(by.css(`[value=${otherType.type}]`)).click();
+    FU.input('$ctrl.otherType', otherType.other);
+    FU.buttons.submit();
+    components.notification.hasSuccess();
+  });
+
+  it('Dont creates a new transaction type for missing type', () => {
+    FU.buttons.create();
+    element(by.model('$ctrl.transactionType.type')).click();
+    element(by.css(`[value=${otherType.type}]`)).click();
+    FU.buttons.submit();
+
+    // check validations
+    FU.validation.error('$ctrl.transactionType.text');
+    FU.validation.error('$ctrl.transactionType.description');
+    FU.validation.error('$ctrl.transactionType.prefix');
+    FU.validation.error('$ctrl.otherType');
+
+    // be sure not success
+    expect(element(by.css('[data-notification-type="notification-success"]')).isPresent())
+      .to.eventually.equal(false);
+
+    FU.modal.cancel();
+
+    components.notification.hasDanger();
+  });
+
+
+  it('Dont creates a new transaction type for missing required values', () => {
+    FU.buttons.create();
+    FU.buttons.submit();
+
+    // check validations
+    FU.validation.error('$ctrl.transactionType.text');
+    FU.validation.error('$ctrl.transactionType.description');
+    FU.validation.error('$ctrl.transactionType.prefix');
+    FU.validation.error('$ctrl.transactionType.type');
+
+    // be sure not success
+    expect(element(by.css('[data-notification-type="notification-success"]')).isPresent())
```

Comment:
See [this comment](https://github.com/IMA-WorldHealth/bhima-2.X/pull/675/files#r76961839)